src/shell/mod.rs

```diff
@@ -774,103 +770,106 @@ fn resolve_command_path(
+  async move {
+    let mut result = Vec::new();
+    let mut current_text = String::new();
+    for part in parts {
+      let evaluation_result_text = match part {
+        WordPart::Text(text) => {
+          current_text.push_str(&text);
+          None
+        }
+        WordPart::Variable(name) => state.get_var(&name).map(|v| v.to_string()),
+        WordPart::Command(list) => Some(
+          evaluate_command_substitution(
+            list,
+            // contain cancellation to the command substitution
+            &state.with_child_token(),
+            stdin.clone(),
+            stderr.clone(),
+          )
+          .await,
+        ),
+        WordPart::Quoted(parts) => {
+          let text =
+            evaluate_word_parts(parts, state, stdin.clone(), stderr.clone())
+              .await
+              .join(" ");
+          current_text.push_str(&text);
+          continue;
+        }
+      };
 
-    // This text needs to be turned into a vector of strings.
-    // For now we do a very basic string split on whitespace, but in the future
-    // we should continue to improve this functionality.
-    if let Some(text) = evaluation_result_text {
-      let mut parts = text
-        .split(' ')
-        .map(|p| p.trim())
-        .filter(|p| !p.is_empty())
-        .collect::<Vec<_>>();
-
-      if !parts.is_empty() {
-        // append the first part to the current text
-        let first_part = parts.remove(0);
-        current_text.push_str(first_part);
-
-        // store the current text
-        result.push(current_text);
-
-        // store all the parts
-        result.extend(parts.into_iter().map(|p| p.to_string()));
-
-        // use the last part as the current text so it maybe
-        // gets appended to in the future
-        current_text = result.pop().unwrap();
+      // This text needs to be turned into a vector of strings.
+      // For now we do a very basic string split on whitespace, but in the future
+      // we should continue to improve this functionality.
+      if let Some(text) = evaluation_result_text {
+        let mut parts = text
+          .split(' ')
+          .map(|p| p.trim())
+          .filter(|p| !p.is_empty())
+          .collect::<Vec<_>>();
+
+        if !parts.is_empty() {
+          // append the first part to the current text
+          let first_part = parts.remove(0);
+          current_text.push_str(first_part);
+
+          // store the current text
+          result.push(current_text);
+
+          // store all the parts
+          result.extend(parts.into_iter().map(|p| p.to_string()));
+
+          // use the last part as the current text so it maybe
+          // gets appended to in the future
+          current_text = result.pop().unwrap();
+        }
       }
     }
+    if !current_text.is_empty() {
+      result.push(current_text);
+    }
+    result
   }
-  if !current_text.is_empty() {
-    result.push(current_text);
-  }
-  result
+  .boxed()
```

Comment:
Shouldn't it be `boxed_local()`?

Reply:
Yup, it would just cause a refactor of some other code (which I just did and will push soon). This was from back when I didn't understand .boxed() vs boxed_local().